docs/data/stashcache/vo-registration.md

```diff
@@ -0,0 +1,147 @@
+  For information about setting up a cache, see the [installation document](/data/stashcache/install-cache/).
+
+
+Choosing namespaces
+-------------------
+
+A namespace should be easy for your users to remember but not so generic that it collides with other VOs.
+We recommend using the lowercase version of your VO as the top-level directory.
+In addition, public data, if any, should be stored in a subdirectory named `PUBLIC`,
+and protected data, if any, should be stored in a subdirectory named `PROTECTED`.
+
+Putting this together, if your VO is named Astro, you should have:
+
+- `/astro/PUBLIC` for public data
+- `/astro/PROTECTED` for protected data
+
+Separating the public and protected data in separate directory trees is preferred for technical reasons.
+
+
+Registering data federation information
+---------------------------------------
+
+In order to declare your VO's role in the StashCache federation,
+you must add StashCache information to the VO's YAML file in the OSG Topology repository.
+
+The data will look like the following:
+```
+DataFederations:
+  StashCache:
+    Namespaces:
+      <Namespaces section>
+    AllowedCaches:
+      <AllowedCaches list>
+    AllowedOrigins:
+      <AllowedOrigins list>
+```
+
+The sections are described below.
+
+
+### Namespaces section
+
+In the namespaces section, you will declare one or more namespaces and, for each namespace,
+list who is allowed to access that namespace.
+
+The list will contain one or more of these:
+
+- `FQAN:<VOMS FQAN>` allows someone using a proxy with the specified VOMS FQAN
+- `DN:<DN>` allows someone using a proxy with that specific DN
+- `ANY` allows anyone; this is used for public data
+
+A complete declaration looks like:
+```
+    Namespaces:
+      /astro/PUBLIC:
+        - ANY
+      /astro/PROTECTED:
+        - FQAN:/Astro
+        - DN:/DC=org/DC=opensciencegrid/O=Open Science Grid/OU=People/CN=Matyas Selmeci
+```
+
+This declares two namespaces: `/astro/PUBLIC` for public data, and `/astro/PROTECTED`
+which can only be read by someone with the `/Astro` FQAN or by Matyas Selmeci.
+
+
+### AllowedCaches list
+
+AllowedCaches is a list of which caches are allowed to host copies of your data.
+There are two cases:
+
+- If you only have public data, your AllowedCaches list can look like:
+
+        AllowedCaches:
+            - ANY
+
+   This allows any cache to host a copy of your data.
+
+- If you have some protected data, then AllowedCaches is a list of _resources_ that are allowed to cache your data.
+   A resource is an entry in a `/topology/<FACILITY>/<SITE>/<RESOURCEGROUP>.yaml` file,
+   for example "CHTC_STASHCACHE_CACHE".
+
+   The following requirements must be met for the resource:
+
+   - It must have an "XRootD cache server" service
+   - It must have an AllowedVOs list that includes either your VO, "ANY", or "ANY_PUBLIC"
+   - It must have a DN attribute with the DN of its host cert
```

Comment:
We should really have a topology page that people can consult to find a list of acceptable caches/origins

Reply:
A form on topology.opensciencegrid.org where you enter your VO and it lists caches and origins that your VO could use.

Reply:
Sounds good. https://opensciencegrid.atlassian.net/browse/SOFTWARE-3550